cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,101 @@
+        }
+        self.postrun_cmds = ['echo $?']
+        self.tags = {'ops', 'diagnostic'}
+        self.maintainers = ['CB']
+
+
+@rfm.parameterized_test(['/scratch/snx1*'],
+                        ['/scratch/snx3*'])
+class FileSystemChangeDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'Change directory to scratch test'
+        self.reference = {
+            'daint:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            },
+            'dom:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time cd'
+        self.executable_opts = [variant]
+
+
+@rfm.parameterized_test(['/scratch/snx1*'],
+                        ['/scratch/snx3*'])
+class FileSystemLsDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'ls of directory in scratch test'
+        self.reference = {
+            'daint:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            },
+            'dom:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time ls'
+        self.executable_opts = [variant]
+
+                       
+@rfm.parameterized_test(['/project/csstaff/bignamic'],#TODO: can we parametrize the user?
+                        ['/users/bignamic'],
+                        ['/scratch/snx1*/bignamic'],
+                        ['/scratch/snx3*/bignamic'])#TODO: what is the correct way to do this?
+class FileSystemDuDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'du of directory'
+
+        # TODO: is it possible to append a pattern?
+        self.perf_patterns = {
+            'real_time': sn.extractsingle(r'\nreal.+m(?P<real_time>\S+)s',
+                                          self.stderr, 'real_time', float),
+#            'size' : sn.extractsingle(r'(?P<size>\S+)\s+%s'%variant,
+#                                      self.stdout, 'size', float)
+            'size' : sn.extractsingle(r'(?P<size>\S+).+/bignamic',# TODO: this should be solved with parametrized user 
+                                      self.stdout, 'size', float)
+        }
+        
+        # TODO: system is not always relevant 
+        self.reference = {
+            '/project/csstaff/bignamic': {
+                'size': (1000, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')                
+            },
+            '/users/bignamic': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            },
+            '/scratch/snx3*/bignamic': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            }
+        }        
+        self.executable = 'time du -mhs --block-size=1M'
```

Comment:
Does it make sense to use full paths like
```
/usr/bin/time
/usr/bin/du
```
?

Reply:
No for `time`, since it's a bash builtin. Yes for `du`.